qiskit_experiments/calibration/experiments/drag.py

```diff
@@ -0,0 +1,438 @@
+                        "freq1": user_bounds.get("freq1", None) or (0, np.inf),
+                        "freq2": user_bounds.get("freq2", None) or (0, np.inf),
+                        "beta": user_bounds.get("beta", None) or (min_beta, max_beta),
+                        "base0": user_bounds.get("base0", None) or (-1 * max_abs_y, 1 * max_abs_y),
+                        "base1": user_bounds.get("base1", None) or (-1 * max_abs_y, 1 * max_abs_y),
+                        "base2": user_bounds.get("base2", None) or (-1 * max_abs_y, 1 * max_abs_y),
+                    },
+                }
+
+                fit_option.update(options)
+                fit_options.append(fit_option)
+
+        return fit_options
+
+    def _post_analysis(self, analysis_result: CurveAnalysisResult) -> CurveAnalysisResult:
+        """Algorithmic criteria for whether the fit is good or bad.
+
+        A good fit has:
+            - a reduced chi-squared lower than three,
+            - a drag parameter value contained in the range of x-values, and
+            - an error on the drag beta smaller than the beta.
+        """
+
+        fit_beta = get_opt_value(analysis_result, "beta")
+        fit_beta_err = get_opt_error(analysis_result, "beta")
+
+        x_data = self._data("series-0").x
+        min_x, max_x = min(x_data), max(x_data)
+
+        criteria = [
+            analysis_result["reduced_chisq"] < 3,
+            min_x <= fit_beta <= max_x,
+            fit_beta_err < abs(fit_beta),
+        ]
+
+        if all(criteria):
+            analysis_result["quality"] = "computer_good"
+        else:
+            analysis_result["quality"] = "computer_bad"
+
+        return analysis_result
+
+
+class DragCal(BaseExperiment):
+    """An experiment that scans the drag parameter to find the optimal value.
+
+    The Drag calibration will run several series of circuits. In a given circuit
+    a xp(β) - xm(β) block is repeated :math:`N` times. As example the circuit of a single
+    repetition, i.e. :math:`N=1`, is shown below.
+
+    .. parsed-literal::
+
+                   ┌───────┐ ░ ┌───────┐ ░ ┌─┐
+              q_0: ┤ xp(β) ├─░─┤ xm(β) ├─░─┤M├
+                   └───────┘ ░ └───────┘ ░ └╥┘
+        measure: 1/═════════════════════════╩═
+                                            0
+
+    Here, the xp gate and the xm gate are intended to be pi and -pi rotations about the
+    x-axis of the Bloch sphere. The parameter β is scanned to find the value that minimizes
+    the leakage to the second excited state. Note that the analysis class requires this
+    experiment to run with three repetition numbers.
+    """
+
+    __analysis_class__ = DragCalAnalysis
+
+    @classmethod
+    def _default_run_options(cls) -> Options:
+        """Default option values for the experiment :meth:`run` method."""
+        return Options(
+            meas_level=MeasLevel.KERNELED,
+            meas_return="single",
```

Comment:
How did you decide on this for the default? Usually you have a decent pi pulse before you calibrate drag so you could discriminate states.

Reply:
Good point I changed to classified here 79ceeb0  and added a test for KERNELED.